homeassistant/components/switchbee/light.py

```diff
@@ -0,0 +1,164 @@
+"""Support for SwitchBee light."""
+
+import logging
+from typing import Any
+
+from switchbee.api import SwitchBeeDeviceOfflineError, SwitchBeeError
+from switchbee.device import ApiStateCommand, DeviceType, SwitchBeeBaseDevice
+
+from homeassistant.components.light import ATTR_BRIGHTNESS, ColorMode, LightEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SwitchBeeCoordinator
+from .const import DOMAIN
+from .entity import SwitchBeeDeviceEntity
+
+MAX_BRIGHTNESS = 255
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def brightness_hass_to_switchbee(value: int) -> int:
+    """Convert hass brightness to SwitchBee."""
+    sb_brightness = int(100 * (value) / MAX_BRIGHTNESS)
+    # SwitchBee maximum brightness is 99
+    return sb_brightness if sb_brightness != 100 else 99
+
+
+def brightness_switchbee_to_hass(value: int) -> int:
+    """Convert SwitchBee brightness to hass."""
+    if value == 99:
+        value = 100
+    return round(value * MAX_BRIGHTNESS / 100)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up SwitchBee light."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        SwitchBeeLightEntity(switchbee_device, coordinator)
+        for switchbee_device in coordinator.data.values()
+        if switchbee_device.type == DeviceType.Dimmer
+    )
+
+
+class SwitchBeeLightEntity(SwitchBeeDeviceEntity, LightEntity):
```

Comment:
I think this needs to be adjusted similar to `SwitchBeeSwitchEntity`, with a local `TypeVar`:
```
class SwitchBeeLightEntity(SwitchBeeDeviceEntity[_DeviceTypeT], LightEntity):
```

Reply:
Why not use `SwitchBeeDimmer` type? this is the only type expected here

Reply:
If it's the only type, then TypeVar isn't needed.
You can use:
```suggestion
class SwitchBeeLightEntity(SwitchBeeDeviceEntity[SwitchBeeDimmer], LightEntity):
```